Tests/AppTests/IngestorTests.swift

```diff
@@ -390,31 +399,51 @@ class IngestorTests: AppTestCase {
             .unwrap()
-        XCTAssertEqual(repos.map(\.$package.id), [try ingested.requireID()])
-
-        // validate packages
-        let reconciled = try await Package.query(on: app.db)
-            .filter(\.$processingStage == .reconciliation)
+        let failed = try await Package.query(on: app.db)
+            .filter(\.$status == .ingestionFailed)
             .first()
             .unwrap()
-        // the ingested package has the update ...
-        XCTAssertEqual(ingested.status, .new)
-        XCTAssertEqual(ingested.processingStage, .ingestion)
-        XCTAssert(ingested.updatedAt! > lastUpdate)
-        // ... while the reconciled package remains unchanged ...
-        XCTAssertEqual(reconciled.status, .new)
-        XCTAssertEqual(reconciled.processingStage, .reconciliation)
-        XCTAssert(reconciled.updatedAt! < lastUpdate)
-        // ... and an error has been logged
+        XCTAssertEqual(succeeded.processingStage, .ingestion)
+        XCTAssertEqual(failed.processingStage, .ingestion)
+        // an error must have been logged
         try logger.logs.withValue { logs in
             XCTAssertEqual(logs.count, 1)
             let log = try XCTUnwrap(logs.first)
             XCTAssertEqual(log.level, .critical)
-            XCTAssertEqual(log.message, #"duplicate key value violates unique constraint "idx_repositories_owner_name""#)
+            XCTAssertEqual(log.message, #"Ingestion.Error(\#(try failed.requireID()), repositorySaveUniqueViolation(duplicate key value violates unique constraint "idx_repositories_owner_name"))"#)
+        }
+
+        // ensure analysis can process these packages
+        try await withDependencies {
+            $0.date.now = .now
+            $0.environment.allowSocialPosts = { false }
+        } operation: { [db = app.db] in
+            Current.fileManager.fileExists = { @Sendable _ in true }
+            Current.git.commitCount = { @Sendable _ in 1 }
+            Current.git.firstCommitDate = { @Sendable _ in .t0 }
+            Current.git.lastCommitDate = { @Sendable _ in .t0 }
+            Current.git.getTags = { @Sendable _ in [] }
+            Current.git.hasBranch = { @Sendable _, _ in true }
+            Current.git.revisionInfo = { @Sendable _, _ in .init(commit: "sha0", date: .t0) }
+            Current.git.shortlog = { @Sendable _ in "" }
+            Current.shell.run = { @Sendable cmd, _ in
+                if cmd.description.hasSuffix("package dump-package") {
+                    return .packageDump(name: "foo")
+                }
+                return ""
+            }
+
+            try await Analyze.analyze(client: app.client, database: db, mode: .id(.id0))
+            try await Analyze.analyze(client: app.client, database: db, mode: .id(.id1))
+            try await XCTAssertEqualAsync(try await Package.find(.id0, on: db)?.processingStage, .analysis)
+            try await XCTAssertEqualAsync(try await Package.find(.id1, on: db)?.processingStage, .analysis)
```

Comment:
This is where we're ensuring analysis can deal with this scenario where one package doesn't have a corresponding repository record.